### DISCUSSION TITLE
Links have a weird behavior when ports are positioned manually

### DISCUSSION THREAD

**Discussion by jaimeatstariongroup** [Blazor-Diagrams/Blazor.Diagrams#283] [General]
Hi, I'm creating a custom node with a shape of a ellipse and placing the ports manually around the perimeter. However the links between the ports have a weird behavior. 

![image](https://user-images.githubusercontent.com/112856984/205963017-2f02249b-a32a-4492-aa09-cbe488de8c16.png)

I don't know how the angles of the target/source are calculated but I thinks it should work something like this. In the image below the vector is calculated from center of the node to the port. I think that should help to display the ports in a better way.

![image](https://user-images.githubusercontent.com/112856984/205964686-d4090ee7-ff81-4f92-b06a-f8d23a639f67.png)

Anyways, how can I achieve a behavior like the described? Because actually the _diagram-svg-layer_ is below the _diagram-html-layer_ so normally you wouldn't see some parts of the links.

**Comment by zHaytam** [comment]
Hello, the control points of the curve are decided using the alignment of the ports. In your example, I believe those ports dont have a BottomLeft alignment, and that's why?

**Reply by jaimeatstariongroup** [reply]
Hi, sorry for the late reply. I am placing the ports manually. But I think the programmer should be able to change the entrance/ exit directions of the ports or at least those directions are calculated in the way I mentioned, this can help to solve these cases.
